logback-core/src/main/java/ch/qos/logback/core/net/ssl/SSLConfigurableSocket.java

```diff
@@ -60,4 +64,28 @@ public void setWantClientAuth(boolean state) {
         delegate.setWantClientAuth(state);
     }
 
+    public void setDisableHostnameVerification(boolean disableHostnameVerification) {
+        if (disableHostnameVerification) {
+            return;
+        }
+        SSLParameters sslParameters = delegate.getSSLParameters();
+        try {
+            /*
+             * SSLParameters.setEndpointIdentificationAlgorithm(...)
+             * is only available from Java 7, so we use reflection
+             * to check whether it is available.
+             */
+            Method setEndpointIdentificationAlgorithmMethod = SSLParameters.class.getMethod("setEndpointIdentificationAlgorithm", String.class);
+            setEndpointIdentificationAlgorithmMethod.invoke(sslParameters, "HTTPS");
+            delegate.setSSLParameters(sslParameters);
+        } catch (NoSuchMethodException e) {
+            // We don't do anything when the method is not available.
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } catch (IllegalArgumentException e) {
+            throw new RuntimeException(e);
```

Comment:
:-)

With the exception of FindBugs in "high priority" mode, I am yet to encounter an automatic tool with a reasonable signal to noise ratio. Codacy does not seem to be an exception to the rule.